tests/conftest.py

```diff
@@ -333,3 +334,34 @@ def func(count, **kwargs):
         return [project_membership(**kwargs) for i in range(count)]
 
     return func
+
+
+@pytest.fixture
+def role_factory():
+    """A fixture for dynamically creating a role with a given set of permissions."""
+
+    def _role_factory(*, permissions):
```

Comment:
The role_factory takes a list of permissions, but in all cases only a single permission is provided. I'm not sure if it's worth continuing accepting a list because it makes semantic sense and might be useful, or if it would be worth removing it and simplifying the code slightly, what do you think?

Reply:
I wasn't sure whether I was going to encounter any cases where we needed a role with more than one permission. I'm still not 🙂